dracut/30ignition/ignition-setup.service

```diff
@@ -8,7 +8,7 @@ Before=local-fs-pre.target
 Before=ignition-disks.service
 Before=ignition-files.service
 
-Requires=dev-disk-by\x2dlabel-boot.device
+Wants=dev-disk-by\x2dlabel-boot.device
```

Comment:
Hm, ok yeah the `After` will still order us.  I wonder whether using `RequiresMountsFor=/boot` would work.  But this looks OK.

Reply:
I guess `RequiresMountsFor` wouldn't work as expected, as it would introduce a dependency to **mounting** the device, while currently is is only waiting for the availability of a device with that label without mounting it.

It seems to me this is intentional - I assume it's expected that Ignition's file stage may change the boot partition itself?

Reply:
> It seems to me this is intentional - I assume it's expected that Ignition's disk stage may change the boot partition itself?

Strictly speaking, yes since Ignition has no differentiation of a boot partition and any other partition. However changing the boot partition probably means the system won't boot anymore since it has things like the grub configs, kernels, etc. Basically Ignition supports it in the same way it supports any other destructive actions that would brick the system.